### DISCUSSION TITLE
Kafka De/Serialization

### DISCUSSION THREAD

**Discussion by bsamartins** [square/wire#2470] [Q&A]
Are there any examples on integrating Wire with Kafka Proto Serializer / Deserializer or at least how to convert from one message type to another?

**Comment by oldergod** [comment]
I don't think there is any. Could you pass bytes around to encode in one and decode in the other, vice-versa? like we do in tests with protoc: https://github.com/square/wire/blob/master/wire-protoc-compatibility-tests/src/test/java/com/squareup/wire/InteropChecker.kt#L94-L101

**Reply by bsamartins** [reply]
Guess there's no way around serializing / deserializing.
Thanks 👍
